src/Commands/MigrateCommands.php

```diff
@@ -0,0 +1,169 @@
+    $this->database = $database;
+    $this->configFactory = $configFactory;
+    $this->keyValue = $keyValue;
+    $this->entityFieldManager = $entityFieldManager;
+    $this->entityLastInstalledSchemaRepository = $entityLastInstalledSchemaRepository;
+    $this->entityDisplayRepository = $entityDisplayRepository;
+    $this->widgetPluginManager = $widgetPluginManager;
+  }
+
+  /**
+   * Migrates an entity_reference field to entity_reference_override.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID.
+   * @param string $field_name
+   *   The field name.
+   *
+   * @command entity_reference_override:migrate
+   *
+   * @usage drush entity_reference_override:migrate
+   *   Migrates an entity_reference field to entity_reference_override.
+   */
+  public function migrate($entity_type_id, $field_name) {
+    $field_storage_config = $this->configFactory->getEditable("field.storage.$entity_type_id.$field_name");
+
+    if ($field_storage_config->get('type') !== 'entity_reference') {
+      $this->io()->error(\dt('Not an entity reference field'));
+      return;
+    }
+
+    $schema_spec = [
+      'description' => 'A map to overwrite entity data per instance.',
+      'type' => 'text',
+      'size' => 'big',
+    ];
+
+    $this->database->schema()->addField($entity_type_id . '__' . $field_name, $field_name . '_overwritten_property_map', $schema_spec);
+    $this->database->schema()->addField($entity_type_id . '_revision__' . $field_name, $field_name . '_overwritten_property_map', $schema_spec);
+
+    $store = $this->keyValue->get("entity.storage_schema.sql");
+    $data = $store->get("$entity_type_id.field_schema_data.$field_name");
+    $data["{$entity_type_id}__$field_name"]['fields']["{$field_name}_overwritten_property_map"] = $schema_spec;
+    $data["{$entity_type_id}_revision__$field_name"]['fields']["{$field_name}_overwritten_property_map"] = $schema_spec;
+    $store->set("$entity_type_id.field_schema_data.$field_name", $data);
+
+    $schema_definitions = $this->entityLastInstalledSchemaRepository->getLastInstalledFieldStorageDefinitions($entity_type_id);
+    $schema_definitions[$field_name]->set('type', 'entity_reference_override');
+    $this->entityLastInstalledSchemaRepository->setLastInstalledFieldStorageDefinitions($entity_type_id, $schema_definitions);
+
+    $this->entityFieldManager->clearCachedFieldDefinitions();
+
+    $field_storage_config->set('type', 'entity_reference_override');
+    $field_storage_config->save(TRUE);
+
+    FieldStorageConfig::loadByName($entity_type_id, $field_name)->calculateDependencies()->save();
+
+    // Use the default widget and settings.
+    $component = $this->widgetPluginManager->prepareConfiguration('entity_reference_override', []);
+
+    $field_map = $this->entityFieldManager->getFieldMapByFieldType('entity_reference')[$entity_type_id][$field_name];
```

Comment:
I considered whether there might be so many that we'd need to batch but I don't think thats the case. I think we could have a thousand old bundle fields and this would still complete in most CLI processes... and no reasonable site will have a thousand entity reference bundle fields... so... this loop looks good to me.